src/validation.cpp

```diff
@@ -565,6 +568,11 @@ static bool CheckInputsFromMempoolAndCache(const CTransaction& tx, CValidationSt
     return CheckInputs(tx, state, view, true, flags, cacheSigStore, true, txdata);
 }
 
+/**
+ * @param[out] coins_to_uncache   Return any outpoints which were not previously present in the
+ *                                coins cache, but were added as a result of validating the tx
+ *                                for mempool acceptance. This is used for DoS protection.
```

Comment:
I suggest removing the last sentence 'This is used for DoS protection.' Without further context, it's not clear how this is used for DoS protection. Your new comment below gives that context and is easy to find (since ATMPW is only called in one place).

Reply:
Changed it to avoid mentioning DoS protection but instead articulate why the caller might make use of it.